### PR TITLE
docs(pattern): generating an API key

### PR DESCRIPTION
**Pull request by connor-leech**
This was a contribution.

Original PR: https://github.com/carbon-design-system/carbon-website-archive/pull/1672

### CODE REVIEW

**Comment by vpicone** [comment]
@connor-leech it looked like it was closed, is there any reason for it to be open here? Trying to avoid a bunch of open issues without any potential for being addressed.

File: src/pages/experimental/generate-an-api-key/index.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+title: Generate an API key
+---
+
+<PageDescription>
+
+Ensure users know what the key is created for, the security implications, and the end destination of the key.
```
Comment (janchild) [inline src/pages/experimental/generate-an-api-key/index.mdx]:
```suggestion
When designing for the generation of API keys, ensure users know what the key is created for, the security implications, and the end destination of the key.
```

File: src/pages/experimental/generate-an-api-key/index.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+[Experimental](/experimental/overview)
+
+#### Maintainer:
+
+[Vikki Paterson](https://github.com/vikkipaterson)
+
+![Example of an API key being generated](images/1.png)
+
+## General access
```
Comment (janchild) [inline src/pages/experimental/generate-an-api-key/index.mdx]:
```suggestion
## General access keys
```

File: src/pages/experimental/generate-an-api-key/index.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+<Column colLg={8}>
+
+![Example of an API key destination selector](images/7.png)
+
+<Caption>Example of an API key destination selector</Caption>
+
+</Column>
+</Row>
+
+## Restricted access
```
Comment (janchild) [inline src/pages/experimental/generate-an-api-key/index.mdx]:
```suggestion
## Restricted access keys
```

File: src/pages/experimental/generate-an-api-key/index.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+[Vikki Paterson](https://github.com/vikkipaterson)
+
+![Example of an API key being generated](images/1.png)
+
+## General access
+
+### Instant generation
+
+Users click a primary ‘Generate’ button. Consider displaying a ‘Generating’ state if the API key takes time to generate.
```
Comment (janchild) [inline src/pages/experimental/generate-an-api-key/index.mdx]:
```suggestion
Users click a primary **Generate** button. Consider displaying a **Generating...** state if the API key takes time to generate.
```

File: src/pages/experimental/generate-an-api-key/index.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+<Row>
+<Column colLg={8}>
+
+![Example of a generate button.](images/2.png)
+
+</Column>
+</Row>
+
+Once the API key is generated it displays in a modal. Include a 'copy' button.
```
Comment (janchild) [inline src/pages/experimental/generate-an-api-key/index.mdx]:
```suggestion
Once the API key is generated it displays in a modal. Include a **Copy** button.
```

**Pull request merged**